tpdatasrc/tpgamefiles/rules/d20_combat/damage_critter.py

```diff
@@ -2,13 +2,140 @@
 import roll_history
 import logbook
 
+debug_enabled = True
+
+def debug_print(*args):
+    if debug_enabled:
+        for arg in args:
+            print arg,
+    return
+
+def playSoundEffect(weaponUsed, attacker, tgt, sound_number):
+    missing_stub("auto soundId = inventory.GetSoundIdForItemEvent(weaponUsed, attacker, tgt, 6); sound tbd")
+    missing_stub("sound.PlaySoundAtObj(soundId, attacker); sound tbd")
+    return
+
 def deal_attack_damage(attacker, tgt, d20_data, flags, action_type): 
     # return value is used by Coup De Grace
-    print('Quel dommage')
-    #...
-    # should call damage_critter() eventually
-    #...
-    return -1
+
+    #Provoke Hostility
+    missing_stub("aiSys.ProvokeHostility(attacker, tgt, 1, 0);")
+
+    #Check if target is alive
+    missing_stub("if critterSys.IsDeadNullDestroyed(tgt), is it enough to just query if dead?")
+    if tgt.d20_query(Q_Dead):
+        return -1
+
+    #Create evt_obj_dam
+    evt_obj_dam = tpdp.EventObjDamage()
+    evt_obj_dam.attack_packet.action_type = action_type
+    evt_obj_dam.attack_packet.attacker = attacker
+    evt_obj_dam.attack_packet.target = tgt
+    evt_obj_dam.attack_packet.event_key = d20_data
+    evt_obj_dam.attack_packet.set_flags(flags)
+
+    #Set weapon used
+    unarmed = OBJ_HANDLE_NULL
+    if evt_obj_dam.attack_packet.get_flags() & D20CAF_TOUCH_ATTACK:
+        evt_obj_dam.attack_packet.set_weapon_used(unarmed)
+    elif evt_obj_dam.attack_packet.get_flags() & D20CAF_SECONDARY_WEAPON:
+        offhandItem = attacker.item_worn_at(item_wear_weapon_secondary)
+        if offhandItem.type != obj_t_weapon:
+            evt_obj_dam.attack_packet.set_weapon_used(unarmed)
+        else:
+            evt_obj_dam.attack_packet.set_weapon_used(offhandItem)
+    else:
+        mainhandItem = attacker.item_worn_at(item_wear_weapon_primary)
+        if mainhandItem.type != obj_t_weapon:
+            evt_obj_dam.attack_packet.set_weapon_used(unarmed)
+        else:
+            evt_obj_dam.attack_packet.set_weapon_used(mainhandItem)
+
+    #Check ammo
+    evt_obj_dam.attack_packet.ammo_item = attacker.get_ammo_used()
+
+    #Check Concealment Miss
+    #Remark: I can't remember that D20CAF_CONCEALMENT_MISS was actually set
+    #in to_hit_processing
+    if flags & D20CAF_CONCEALMENT_MISS:
+        missing_stub("histSys.CreateRollHistoryLineFromMesfile(11, attacker, tgt); to what mesfile this is referring to?")
```

Comment:
history.mes
Note that it might have not been extracted by Co8 since they couldn't invoke this

Reply:
> history.mes
> Note that it might have not been extracted by Co8 since they couldn't invoke this
Ah I see, thanks for pointing this out to me. May I use freeform and just copy it from history.mes then?

Reply:
Definitely not!